src/NetMQ/Core/YPipe.cs

```diff
@@ -180,17 +180,25 @@ public bool CheckRead()
         /// <returns><c>true</c> if the read succeeded, otherwise <c>false</c>.</returns>
         public bool TryRead([MaybeNullWhen(returnValue: false)] out T value)
         {
-            // Try to prefetch a value.
-            if (!CheckRead())
+            try
+            {
+                // Try to prefetch a value.
+                if (!CheckRead())
+                {
+                    value = default(T);
+                    return false;
+                }
+
+                // There was at least one value prefetched.
+                // Return it to the caller.
+                value = m_queue.Pop();
+                return true;
+            }
+            catch
             {
                 value = default(T);
                 return false;
             }
-
-            // There was at least one value prefetched.
-            // Return it to the caller.
-            value = m_queue.Pop();
-            return true;
```

Comment:
Adding `try/catch` can hurt performance. Catching all exceptions raises eyebrows. While this is defensive, I'd like to understand what the actual issue is in order to fix it. Is there a test we could write that reproduces this behaviour, for example?